### PR TITLE
PICARD-1861: Add Configuration Management, Version Compatibility, and Local Source Support to Plugin v3

### PR DESCRIPTION
**Pull request by knguyen1**
# Summary

* This is a…
  * [ ] Bug fix
  * [x] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**: This PR adds three core features to the Plugin v3 system: configuration management for plugin enable/disable state, robust version compatibility checking, and local plugin source implementation with file copying and conflict resolution.

# Problem

* JIRA ticket (_optional_): PICARD-1861

The Plugin v3 system had incomplete TODO items that needed to be resolved:
- TODO in `manager.py:75` for version compatibility checking
- TODO in `plugin.py:85` for local plugin source implementation
- Missing configuration management for plugin state persistence

These features are essential for a complete Plugin v3 system that can safely load plugins, manage their state, and support local development workflows.

# Solution

## Configuration Management
- Added `enabled_plugins3` configuration option to `picard/options.py`
- Implemented plugin state persistence in `PluginManager` with methods:
  - `get_enabled_plugins()`, `set_plugin_enabled()`, `is_plugin_enabled()`
  - `enable_plugin()`, `disable_plugin()` with proper error handling
- Added v2 to v3 plugin configuration migration in `migrate_v2_plugin_config()`
- Enhanced plugin loading to respect configuration state during initialization

## Version Compatibility
- Completed TODO in `manager.py:75` by implementing robust version checking
- Added `_is_plugin_compatible()` function for clean boolean compatibility checks
- Enhanced `_compatible_api_versions()` with proper documentation
- Added detailed error messages showing required vs supported API versions
- Enhanced logging for debugging compatible plugin versions

## Local Plugin Source
- Implemented `PluginSourceLocal.sync()` method with comprehensive file copying
- Added conflict resolution for existing plugin installations (clean reinstall approach)
- Proper handling of symbolic links and file permissions via `shutil.copytree`
- Smart filtering of development artifacts (__pycache__, .git, .pytest_cache, etc.)
- Robust error handling with `PluginSourceSyncError` for sync failures

## Testing
- **61 comprehensive tests** covering all new functionality
- Parameterized tests for edge cases and error scenarios
- Integration tests for full plugin workflow
- Error handling and exception testing
- All tests pass and pre-commit hooks are satisfied

## Technical Considerations
- All changes are additive and backward compatible
- Proper type hints throughout (using `|` union syntax instead of deprecated `typing` aliases)
- NumPy-style docstrings for all public methods
- Exception handling follows project patterns (specific exceptions, not broad catches)
- Uses `contextlib.suppress` for exception suppression where appropriate

# Action

Additional actions required:
* [ ] Update Picard [documentation](https://github.com/metabrainz/picard-docs) (please include a reference to this PR)
* [ ] Other (please specify below)

This PR completes the core Plugin v3 functionality and resolves the outstanding TODO items. The implementation is ready for review and testing.

### CODE REVIEW

File: picard/plugin3/manager.py [hunk 1]
```diff
@@ -26,66 +26,183 @@
...
+    def disable_plugin(self, plugin_name: str) -> None:
+        """Disable a plugin and unload it if loaded."""
+        self.set_plugin_enabled(plugin_name, False)
+        # Find and unload the plugin if it exists
+        plugin_found = False
+        for plugin in self._plugins:
+            # Only unload if already loaded
+            if plugin.name == plugin_name and getattr(plugin, '_module', None):
+                plugin_found = True
+                try:
```
Comment (knguyen1) [inline picard/plugin3/manager.py:158]: see review by @zas: https://github.com/phw/picard/pull/24#discussion_r2344872294

> An helper method could help to reduce code redundancy in enable_plugin() and disable_plugin(), something like:
```python
def find_plugin_by_name(self, plugin_name: str) -> Plugin| None::
    for plugin in self._plugins:
        if plugin.name == plugin_name and getattr(plugin, '_module', None):
            return plugin
    return None
```
